c2rust-ast-exporter/src/AstExporter.cpp

```diff
@@ -2417,22 +2421,44 @@ class TranslateConsumer : public clang::ASTConsumer {
             //
             // Getting all comments requires -fparse-all-comments (see
             // augment_argv())!
+            const SourceManager& sourceMgr = Context.getSourceManager();
+#if CLANG_VERSION_MAJOR < 10
             auto comments = Context.getRawCommentList().getComments();
             cbor_encoder_create_array(&outer, &array, comments.size());
             for (auto comment : comments) {
                 CborEncoder entry;
                 cbor_encoder_create_array(&array, &entry, 4);
 #if CLANG_VERSION_MAJOR < 8
                 SourceLocation loc = comment->getLocStart();
-#else
+#else // 7 < CLANG_VERSION_MAJOR < 10
                 SourceLocation loc = comment->getBeginLoc();
-#endif // CLANG_VERSION_MAJOR
+#endif // CLANG_VERSION_MAJOR < 10
                 visitor.encodeSourcePos(&entry, loc); // emits 3 values
-                auto raw_text = comment->getRawText(Context.getSourceManager());
+                auto raw_text = comment->getRawText(sourceMgr);
                 cbor_encode_byte_string(&entry, raw_text.bytes_begin(),
                                         raw_text.size());
                 cbor_encoder_close_container(&array, &entry);
             }
+#else  // CLANG_VERSION_MAJOR >= 10
+            const FileID file = sourceMgr.getMainFileID();
+            auto comments = Context.getRawCommentList().getCommentsInFile(file);
+            if (comments != nullptr) {
+                cbor_encoder_create_array(&outer, &array, comments->size());
```

Comment:
This looks identical to the code of the other branch, you could probably de-duplicate it using a function or a closure.

Reply:
It is close but not identical. Example:
`auto raw_text = comment->getRawText(sourceMgr);` vs
`auto raw_text = comment.second->getRawText(sourceMgr);`
I had to use so many preprocessor directives that the code became unreadable. I figured we could live with the duplication until we drop support for anything before LLVM 8 and thus can get rid of a lot of the preprocessor definitions.